packages/examples/src/diffie-hellman/diffie-hellman.ts

```diff
@@ -0,0 +1,164 @@
+import { Choreography, Located } from "@choreography-ts/core";
+import { ExpressBackend } from "@choreography-ts/backend-express";
+
+// Typescript implementation of the Diffie-Hellman key exchange algorithm
+// Ported from the HasChor implementation here: https://github.com/gshen42/HasChor/blob/42ae1ef9a500dadd82f0cfe5dee3c2aa631d8f4d/examples/diffiehellman/Main.hs
+
+//////////////////////
+// Helper functions //
+/////////////////////
+
+// Return all divisors of `x`
+const divisors = (x: number): number[] => {
+  let divs: number[] = [1, x];
+  for (let y = 2; y <= x / 2; y++) {
+    if (x % y == 0) divs.push(y);
+  }
+  return divs;
+};
+
+// Checks if input number is prime
+const isPrime = (x: number): boolean => {
+  let divs: number[] = divisors(x);
+  if (divs.length == 2) return true;
+  return false;
+};
+
+// Prime number generator
+const primeNums = function* () {
+  let y = 1;
+  while (y++ > 0) {
+    if (isPrime(y)) yield y;
+  }
+};
+
+// Return random integer in range [min, max]
+// https://stackoverflow.com/a/7228322
+const randInRange = (min: number, max: number) =>
+  Math.floor(Math.random() * (max - min + 1) + min);
+
+///////////////////////////////////
+// Diffie-Hellman implementation //
+///////////////////////////////////
+const locations = ["alice", "bob"];
+export type Location = (typeof locations)[number];
+
+export const diffieHellman = <A extends Location, B extends Location>(
+  a: A,
+  b: B
+) => {
+  const choreo: Choreography<
+    Location,
+    [Located<boolean, A>], // Specify boolean argument specifically for location "A" ("A|B" instead would mean for both)
+    [Located<number, A>, Located<number, B>]
+  > = async ({ locally, comm }, [arg]) => {
+    // The `arg` specifies whether `a` should wait to initiate key exchange
+
+    // Wait for alice to start key exchange (if desired)
+    await locally(a, async (unwrap: (arg: Located<boolean, A>) => boolean) => {
```

Comment:
The type of `unwrap` should be inferred and should not be specified here. The expected behavior is `unwrap` gets `Unwrap<A>`. I want to figure this out before we merge this PR.

```suggestion
    await locally(a, async (unwrap) => {
```

Reply:
Ok I just pushed fixes for this